### DISCUSSION TITLE
Encapsulating of x-model returns Unexpected token '{'

### DISCUSSION THREAD

**Discussion by zraly** [alpinejs/alpine#2881] [7. Bugs (archived)]
Documentation says you can _encapsulate entire Alpine template directives using x-bind_, but I am not able to encapsulate `x-model` directive, as you can see in this fiddle: https://jsfiddle.net/r9zhyu86/2/

There are two examples. First from the documentation of `x-model`, second the same, but encapsulated. 

The second example returns error 

```
Unexpected token '{'

Expression: "['x-model']() {
        return this.message
      } = rightSideOfExpression($event, ['x-model']() {
        return this.message
      })"
```

I also added a button, which clears the message, because it shows, that encapsulated `x-model` still partially works – it updates the message variable in the input.

**Comment by hkhanna** [comment]
I'm having this issue too. At first, I thought I was using Alpine wrong, but I think it's a bug!

**Comment by punkfairie** [comment]
I'm having this issue as well, even with a simple return statment:
```
['x-model']() {
    return this.selectKeys;
},
```

I've got several other bound directives that work perfectly.

**Comment by ekwoka** [comment]
Did you try returning just a string representation of the key?

I think it's just an issue for the two way data-binding. Returning the thing as that does would not allow it to be two way bound.
